src/compiler/parser.ts

```diff
@@ -4778,11 +4778,12 @@ namespace ts {
         function parseCatchClause(): CatchClause {
             const result = <CatchClause>createNode(SyntaxKind.CatchClause);
             parseExpected(SyntaxKind.CatchKeyword);
-            if (parseExpected(SyntaxKind.OpenParenToken)) {
+
+            if (parseOptional(SyntaxKind.OpenParenToken)) {
                 result.variableDeclaration = parseVariableDeclaration();

```

Comment:
Conditionally setting `result.variableDeclaration` introduces polymorphism as we now have two `CatchClause` shapes, one with the property and one without. This can cause performance degradation in V8 (NodeJS). I would recommend you add an `else` clause that sets `result.variableDeclaration = undefined` to keep the shape of `CatchClause` the same regardless of which form the user writes.